firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -161,20 +161,34 @@ public FirebaseFirestoreSettings getFirestoreSettings() {
    */
   @PublicApi
   public void setFirestoreSettings(@NonNull FirebaseFirestoreSettings settings) {
-    checkNotNull(settings, "Provided settings must not be null.");
-    // As a special exception, don't throw if the same settings are passed repeatedly. This
-    // should make it simpler to get a Firestore instance in an activity.
-    if (client != null && !this.settings.equals(settings)) {
-      throw new IllegalStateException(
-          "FirebaseFirestore has already been started and its settings can no longer be changed. "
-              + "You can only call setFirestoreSettings() before calling any other methods on a "
-              + "FirebaseFirestore object.");
+    synchronized (databaseId) {
```

Comment:
Please add a comment to the databaseId field that says mentions that it's being used as a private lock object for this instance. There's nothing inherent about databaseId itself that needs locking, it just saves us creating a separate object that would otherwise looked like this:

```java
private final Object lock = new Object();
```

Reply:
done